python/mlc_chat/cli/chat.py

```diff
@@ -0,0 +1,97 @@
+"""A command line tool for directly chatting with open source LLMs"""
+import os
+
+from mlc_chat import ChatConfig, ChatModule
+from mlc_chat.callback import StreamToStdout
+
+from ..support.argparse import ArgumentParser
+
+
+def print_available_commands():
+    """Function printing available internal commands."""
+    print("You can use the following special commands:")
+    print("\t/help\t\tprint the special commands")
+    print("\t/exit\t\tquite the cli")
+    print("\t/stats\t\tprint out the latest stats (token/sec)")
+    print("\t/reset\t\trestart a fresh chat")
+    print(
+        "\t/reload [model]\treload model `model` from disk, or reload\
+             the current model if `model` not not specified"
+    )
+
+
+def chat(modelname: str, devicename: str, chatconfig: ChatConfig, modellib: str):
+    """Command line chat with a specific open source LLM."""
+    cm = ChatModule(
+        model=modelname, device=devicename, chat_config=chatconfig, model_lib_path=modellib
+    )
+
+    print("Using MLC config: ", os.path.abspath(cm.config_file_path))
+    print("Using model weights:", os.path.abspath(cm.model_path))
+    print("Using model library:", os.path.abspath(cm.model_lib_path))
+
+    print_available_commands()
+    ps = cm._get_role_0()  # pylint: disable=W0212
+
+    prompt = input(f"{ps} ")
+
+    while prompt != "/exit":
+        match prompt:
+            case "/help":
+                print_available_commands()
+            case "/stats":
+                print(f"Statistics: {cm.stats()}\n")
+            case "/reset":
+                cm.reset_chat()
+            case "/reload":
+                print("not yet implemented.")
+            case _:
+                cm.generate(
+                    prompt,
+                    progress_callback=StreamToStdout(callback_interval=2),
+                )
```

Comment:
```suggestion
        if prompt == "/help":
            print_available_commands()
        elif prompt == "/stats":
            print(f"Statistics: {cm.stats()}\n")
        elif prompt == "/reset":
            cm.reset_chat()
        elif prompt == "/reload":
            print("not yet implemented.")
        else:
            cm.generate(
                prompt,
                progress_callback=StreamToStdout(callback_interval=2),
            )
```

Sorry, the python version of our ci is very old, py3.8. It doesn't support match syntax.